src/plugins/share/public/persistable_state/types.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SavedObjectAttributes, SavedObjectReference } from 'src/core/public';
+
+export interface PersistableState<
+  S extends SavedObjectAttributes,
```

Comment:
maybe this should not be refering to savedObjectAttributes ? there might be no saved object asociated with that state, for example saved visualization contains filters (which belong to data plugin, but are not saved on their own)

Reply:
Yeah I agree this is misleading. I used `SavedObjectAttributes` as a convenience because the type essentially allows for any serializable json, but should probably re-write the correct interface here to avoid confusion